### ISSUE TITLE
Exception when saving product

### ISSUE THREAD

**Issue by xtremevision** [mrkhoa99/Boolfly_ProductRelation#13]
Brand new install M2 2.3.1. Tried to create the very first product. Save crashes with error 500.

[Thu Nov 21 15:45:56.161981 2019] [php7:error] [pid 11386] [client 10.1.10.177:62499] PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Magento\\InventoryGroupedProductAdminUi\\Plugin\\Catalog\\Model\\Product\\Link\\ProcessSourceItemsAfterSaveAssociatedLinks::afterSaveProductRelations() must be an instance of Magento\\Catalog\\Model\\Product\\Link, null given, called in /data/vhosts/mike.m2.test/htdocs/vendor/magento/framework/Interception/Interceptor.php on line 146 and defined in /data/vhosts/mike.m2.test/htdocs/vendor/magento/module-inventory-grouped-product-admin-ui/Plugin/Catalog/Model/Product/Link/ProcessSourceItemsAfterSaveAssociatedLinks.php:51\nStack trace:\n#0 /data/vhosts/mike.m2.test/htdocs/vendor/magento/framework/Interception/Interceptor.php(146): Magento\\InventoryGroupedProductAdminUi\\Plugin\\Catalog\\Model\\Product\\Link\\ProcessSourceItemsAfterSaveAssociatedLinks->afterSaveProductRelations(Object(Boolfly\\ProductRelation\\Model\\Catalog\\Product\\Link\\Interceptor), NULL, Object(Boolfly\\ProductRelation\\Model\\Product\\Interceptor))\n#1 /data/vhosts/mike.m2.test/htdocs/vendor/magento/framework/Interception/Interc in /data/vhosts/mike.m2.test/htdocs/vendor/magento/module-inventory-grouped-product-admin-ui/Plugin/Catalog/Model/Product/Link/ProcessSourceItemsAfterSaveAssociatedLinks.php on line 51, referer: http://mikem2test.local/admin/catalog/product/new/set/4/type/simple/key/bf61d9a5d77d9a4718c4fcf3e242d1a1384006a5f030b28415560687fae4c1c1/

**Comment by TommyKolkman** [comment]
Hi @xtremevision, have you ever found a way around this? Experiencing this too, at the moment.

**Comment by xtremevision** [comment]
Frankly I don't remember, more than likely I did, but I don't think I have that instance anymore and I don't know what state it was left in. Why? You have the same problem?

**Comment by TommyKolkman** [comment]
Yes, on a 2.4.2. When I turn off Magento_InventoryConfigurableProductAdminUi it works, but that one is now in a default Magento, so I have to make it work somehow.

**Comment by TommyKolkman** [comment]
If someone ever finds this, I PR'ed it, wasn't too hard.
